pkg/kvstore/allocator/allocator_test.go

```diff
@@ -262,6 +263,10 @@ func (s *AllocatorSuite) TestRunLocksGC(c *C) {
 func (s *AllocatorSuite) TestGC(c *C) {
 	allocatorName := randomTestName()
 	maxID := idpool.ID(256 + c.N)
+
+	option.Config.IdentityStart = 0
+	option.Config.IdentityEnd = int(maxID)
```

Comment:
This unit test modifies the global `option.Config` variable, which means that it will conflict with other unit tests. The range of identities should probably be parameters to `NewKVStoreBackend` instead.

Reply:
@twpayne  Thanks for reviewing! Yes, this unit test will conflict with others, even if the options are restored after the test finishes.

Regard to your suggestion:

> The range of identities should probably be parameters to NewKVStoreBackend instead.

my understanding is:

1. add `idRangeStart` and `idRangeEnd` to [`struct kvstoreBackend`](https://github.com/cilium/cilium/blob/v1.9.5/pkg/kvstore/allocator/allocator.go#L74), 
2. respect this limit when creating an allocator in [`NewAllocator()`](https://github.com/cilium/cilium/blob/v1.9.5/pkg/allocator/allocator.go#L273),
3. if `WithMin()` and/or `WithMax()` are provided to `NewAllocator()`, we should use the **intersection** of the two ranges.

But it should be noted that in this case, **we're assuming `allocator:backend == 1:1`** (seems not a problem).